src/main/java/me/zeroeightsix/kami/module/modules/render/ChunkFinder.java

```diff
@@ -86,14 +109,246 @@ protected void onDisable() {
+                e.printStackTrace();
+                KamiMod.log.error("some exception happened when getting canonicalFile -> " + e.getMessage());
+                Command.sendChatMessage("onGetPath: " + e.getMessage());
+            }
+
+            // Gets the "depth" of this directory relative the the game's run directory, 2 is the location of the world
+            if (rV.toPath().relativize(mc.gameDir.toPath()).getNameCount() != 2) {
+                // subdirectory of the main save directory for this world
+
+                rV = rV.getParentFile();
+            }
+
+        } else { // Otherwise, the server must be remote...
+            rV = makeMultiplayerDirectory();
+        }
+
+        // We will actually store the world data in a subfolder: "DIM<id>"
+        if(dimension != 0) { // except if it's the overworld
+            rV = new File(rV, "DIM" + dimension);
+        }
+        
+        // maybe we want to save it in region folder
+        if(saveInRegionFolder.getValue()) {
+            rV = new File(rV, "region");
+        }
+        
+        rV = new File(rV, "newChunkLogs");
+
+
+        String date = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss").format(new Date());
+        rV = new File(rV, mc.getSession().getUsername() + "_" + date + ".csv"); // maybe dont safe the name actually. But I also dont want to make another option...
+        return rV;
+    }
+    
+    private File makeMultiplayerDirectory(){
+        File rV = Minecraft.getMinecraft().gameDir;
+        String folderName;
+        switch(saveOption.getValue()){
+                case liteLoaderWdl: // make folder structure like liteLoader
+                    folderName = mc.getCurrentServerData().serverName;
+                    
+                    rV = new File(rV, "saves");
+                    rV = new File(rV, folderName);
+                    break;
+                case nhackWdl: // make folder structure like nhack-insdustries
+                    folderName = getNHackInetName();
+                    
+                    rV = new File(rV, "config");
+                    rV = new File(rV, "wdl-saves");
+                    rV = new File(rV, folderName);
+
+                    // extra because name might be different
+                    if (!rV.exists()) {
+                        Command.sendChatMessage("nhack directory doesnt exist: " + folderName);
+                    }
+                    break;
+                default: // make folder structure in .minecraft
+                    folderName = mc.getCurrentServerData().serverName + "-" + mc.getCurrentServerData().serverIP;
+                    if (SystemUtils.IS_OS_WINDOWS) {
+                        folderName = folderName.replace(":", "_");
+                    }
+
+                    rV = new File(rV, "KAMI_NewChunks");
+                    rV = new File(rV, folderName);
+            }
+        
+        return rV;
+    }
+
+    private String getNHackInetName() {
```

Comment:
What's this got to do with nhack?

Reply:
nhack saves its world downloads in different folders then world downloader eg.
Somehow it weirdly converts the ip there. Shall I remove the method and move the code into the switch case?

Reply:
Oh, I see. There's no need to change this.

Reply:
Leave a comment so we know where it came from in 9 months

Reply:
> Leave a comment so we know where it came from in 9 months

Not needed. The code explains itself and where it came from isn't important.